### PR TITLE
Feature/harmony provider mvp

### PR DESCRIPTION
**Pull request by fersirni**

### CODE REVIEW

File: packages/harmony-provider/README.md [hunk 1]
```diff
@@ -0,0 +1,131 @@
+# Harmony SDK
```
Comment (0xslipk) [inline packages/harmony-provider/README.md]: Harmony RPC Providers

File: packages/harmony-provider/README.md [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+
+Harmony SDK provides a collection of methods to interact with the Harmony blockchain.
+
+Currently under developement 🤓
+
+## Install
+
+```sh
+npm i harmony-sdk
```
Comment (0xslipk) [inline packages/harmony-provider/README.md]: `npm i @harmony-js/providers`

File: packages/harmony-provider/README.md [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+```
+
+## Usage
+
+### Node
+
+```typescript
+// JavaScript
+const { HarmonyProvider } = require('harmony-sdk')
```
Comment (0xslipk) [inline packages/harmony-provider/README.md]: const { HarmonyProvider } = require('@harmony-js/providers')

File: packages/harmony-provider/README.md [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+
+// TypeScript
+import { HarmonyProvider } from 'harmony-sdk'
```
Comment (0xslipk) [inline packages/harmony-provider/README.md]: import { HarmonyProvider } from '@harmony-js/providers'

File: packages/harmony-provider/README.md [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+| Method | Description |
+| ------------- | ------------- |
+| getBlockNumber | Gets the last block number. |
+| getBlock | Gets the block data by block number or block hash. |
+| getTransaction | Gets the transaction data by transaction hash. |
+
+### Initializing
+
+```typescript
+import { HarmonyProvider, HMY_TESTNET_NETWORK } from 'harmony-sdk'
```
Comment (0xslipk) [inline packages/harmony-provider/README.md]: import { HarmonyProvider, HMY_TESTNET_NETWORK } from '@harmony-js/providers'

File: packages/harmony-provider/README.md [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+```
+
+### Methods
+
+#### getBlockNumber
+
+Returns the last block number on the blockchain.
+
+```ts
+import { HarmonyProvider, HarmonyProviderMethods, HMY_TESTNET_NETWORK } from 'harmony-sdk'
```
Comment (0xslipk) [inline packages/harmony-provider/README.md]: import { HarmonyProvider, HarmonyProviderMethods, HMY_TESTNET_NETWORK } from '@harmony-js/providers'

File: packages/harmony-provider/README.md [hunk 7]
```diff
@@ -0,0 +1,131 @@
...
+const blockNumber = await provider.getBlockNumber()
+// Executing the perform method 
+const otherWayToGetBlockNumber = await provider.perform(HarmonyProviderMethods.getBlockNumber, {})
+```
+#### getBlockByTagOrHash
+
+Returns the block data. The block can be searched by it's hash or number. It can retrieve the block transactions data or just the transactions hashes.
+
+```ts
+import { HarmonyProvider, HarmonyProviderMethods, HMY_TESTNET_NETWORK } from 'harmony-sdk'
```
Comment (0xslipk) [inline packages/harmony-provider/README.md]: same!

File: packages/harmony-provider/src/provider.ts [hunk 8]
```diff
@@ -0,0 +1,134 @@
...
+
+    throw Error(`unsupported network, ${network}`)
+  }
+
+  getNextId(): number {
+    this._nextId += 1
+    return this._nextId
+  }
+
+  async send(method: string, params: Record<string, any> | any[]): Promise<any> {
```
Comment (0xslipk) [inline packages/harmony-provider/src/provider.ts]: why do you implemented send again if this is already implemented in https://github.com/ethers-io/ethers.js/blob/master/packages/providers/src.ts/json-rpc-provider.ts#L450? This code of EXPERIMENTAL_genesis_config just work for near

**Review by 0xslipk — CHANGES_REQUESTED** [review]

File: packages/harmony-provider/src/provider.ts [hunk 9]
```diff
@@ -0,0 +1,77 @@
...
+      default:
+        return super.perform(method, params)
+    }
+  }
+
+  async getTransaction(hash: string): Promise<TransactionResponse> {
+    return this.send('hmy_getTransactionByHash', [hash])
+  }
+
+  async getBlockByTagOrHash(params: blockParams, includeTransactions = false): Promise<any> {
```
Comment (bzuker) [inline packages/harmony-provider/src/provider.ts]: I would call this `getBlock` and try to conform to the interface from `ethers.js`:
```
getBlock(blockHashOrBlockTag: BlockTag | string | Promise<BlockTag | string>): Promise<Block>
```

    
 We can also include another method `getBlockWithTransactions` so that we avoid the boolean flag, which could be hard to parse

File: packages/harmony-provider/src/provider.ts [hunk 10]
```diff
@@ -0,0 +1,77 @@
...
+const HARMONY_MAINNET_URL = 'https://api.harmony.one'
+const HARMONY_TESTNET_URL = 'https://api.s0.b.hmny.io'
+
+export enum HarmonyProviderMethods {
+  getBlockNumber = 'getBlockNumber',
+  getBlock = 'getBlock',
+  getTransaction = 'getTransaction',
+}
+
+interface blockParams {
```
Comment (bzuker) [inline packages/harmony-provider/src/provider.ts]: nit: Let's use CamelCase for interfaces (eg:`BlockParams`)

**Review by bzuker — COMMENTED** [review]
looking good! Left a comment regarding getBlock. 
If possible, let's try to extend ethersjs `Block` so that it conforms Harmony's idea of Block as well and so that we can type the return values

File: packages/harmony-provider/package.json [hunk 11]
```diff
@@ -0,0 +1,34 @@
...
+    "eslint": "^8.11.0",
+    "eslint-config-prettier": "^8.5.0",
+    "eslint-plugin-import": "^2.25.4",
+    "eslint-plugin-prettier": "^4.0.0",
+    "mocha": "^9.2.2",
+    "ts-node": "^10.7.0"
+  },
+  "dependencies": {
+    "@ethersproject/networks": "^5.6.1",
+    "@ethersproject/properties": "^5.6.0",
```
Comment (0xslipk) [inline packages/harmony-provider/package.json]: You are not using this package anywhere.

File: packages/harmony-provider/package.json [hunk 12]
```diff
@@ -0,0 +1,34 @@
...
+    "@ethersproject/providers": "^5.6.2",
+    "@ethersproject/web": "^5.6.0",
```
Comment (0xslipk) [inline packages/harmony-provider/package.json]: You are not using this package anywhere.

File: packages/harmony-provider/package.json [hunk 13]
```diff
@@ -0,0 +1,34 @@
...
+    "typescript": "^4.6.3"
```
Comment (0xslipk) [inline packages/harmony-provider/package.json]: Move ts to the devDependencies

**Review by 0xslipk — APPROVED** [review]
a minor comments.

**Review by 0xslipk — APPROVED** [review]
